Reportedly/Services/Network/RESTful/API/AuthenticationAPIService.swift

```diff
@@ -21,35 +21,43 @@ final class AuthenticationAPIService: BaseAPIService, AuthenticationAPIServicePr
     
     static let shared = AuthenticationAPIService()
 
-    var token: String { tokenData?.value ?? "" }
+    var token: String {
+        get {
+            UserManager.shared.token?.value ?? ""
+        }
+        set {
+            UserManager.shared.token = Token(value: newValue, isExpired: false)
+        }
+    }
 
     // MARK: - Private Variables
     
-    // TODO: - Removed this hard-coded authentication token when server support user's token
-    private var tokenData: Token? = Token(
-        value: "eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMiIsInNjcCI6InVzZXIiLCJhdWQiOm51bGwsImlhdCI6MTYwNzMzMjUyNCwiZXhwIjoxNjA3NDE4OTI0LCJqdGkiOiI0NzVmNzE5ZS02ZTNmLTRmYTUtYjI3Zi02ZDkxMGM5NGRiMzgifQ.l9mFj8r8tfjk0oVyBPKy5HEvBItFzZ-iIlAjcAmKuhA",
-        isExpired: true
-    )
 
     // MARK: - Public Functions
     
-    func login(with loginRequest: LoginRequest, completion: @escaping ResultRestfulCompletion) {
+    func login(with loginRequest: LoginRequest, completion: @escaping ResultCompletion<User>) {
         request(
             topic: RequestResourceType.login.rawValue,
             method: .post,
             bodyParams: loginRequest.toDictionary(),
             shouldAuthenticate: false
         ) { data, success, error in
-            Thread.executeOnMainThread {
-                if let error = error {
-                    return completion(.failure(error))
-                }
-                guard let success = success else {
-                    return completion(.failure(ResponseError(.other)))
-                }
-                switch success.type {
-                case .noContent: completion(.success(ResponseSuccess(.noContent)))
-                case .success: completion(.success(ResponseSuccess(.success)))
+            if let error = error {
+                return Thread.executeOnMainThread { completion(.failure(error)) }
+            }
+            guard let success = success else {
+                return Thread.executeOnMainThread { completion(.failure(ResponseError(.other))) }
+            }
+            switch success.type {
+            case .noContent: Thread.executeOnMainThread { completion(.failure(ResponseError(.wrongJsonFormat))) }
+            case .success:
+                if let data = data, let dataString = String(data: data, encoding: .utf8),
+                   let userDict = dataString.toJSONDictionary?["data"] as? JSONDictionary,
+                   let userData = try? JSONSerialization.data(withJSONObject: userDict, options: .prettyPrinted),
+                   let user: User = try? JSONDecoder().decode(User.self, from: userData) {
```

Comment:
I think we can wrap model user data, so we can omit parsing data here. What do you think? 😅

Reply:
yeah I think your point is good, let me update